### PR TITLE
V1.1 backports 18 08 28 v2

### PR DESCRIPTION
**Pull request by jrfastab**
#5300
#5284
#5252
#5250
#5230
#5206

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5443)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
bisecting don't merge....

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by tgraf** [comment]
@jrfastab Are you happy with the CI runs? Merge it?

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
@tgraf one more run lets see if it still passes. I expect it should.

**Comment by jrfastab** [comment]
@tgraf with the patch https://github.com/cilium/cilium/pull/5443/commits/1c54857162c9a6f4ceb95849e4bfaab57566c49a I see the error here, https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/6183/
```
16:33:34 [K8s-1.10] 
16:33:34 [K8s-1.10]     cilium was not able to get into ready state
16:33:34 [K8s-1.10]     Expected
16:33:34 [K8s-1.10]         <*errors.errorString | 0xc4200892d0>: {
16:33:34 [K8s-1.10]             s: "Timeout reached: could not get Pods",
16:33:34 [K8s-1.10]         }
16:33:34 [K8s-1.10]     to be nil
16:33:34 [K8s-1.10] 
```
If anything comes to mind let me know. I'm going to run it again to see if its a flake.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
@tgraf lets merge this PR as-is and I'll debug the three patches causing issues in another PR.

File: pkg/endpoint/bpf.go [hunk 1]
```diff
@@ -712,6 +712,20 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, boo
 		}
 	}
 
+	// Walk the L4Policy to add new redirects and update the desired policy map
+	// state to set the newly allocated proxy ports.
+	if e.DesiredL4Policy != nil {
+		desiredRedirects, err = e.addNewRedirects(owner, e.DesiredL4Policy)
```
Comment (rlenglet) [inline pkg/endpoint/bpf.go:718]: I think this commit should be reverted. With this, we're calling `addNewRedirects` twice. Here, and on line 700 above. I will send a PR to revert.

Comment (rlenglet) [reply pkg/endpoint/bpf.go:718]: See https://github.com/cilium/cilium/pull/5481.

**Pull request merged**